From 1c539d9d22f20cacb9235a7cc3098e1d5e6bc92a Mon Sep 17 00:00:00 2001
From: Hongbo Wang <hongbo.wang@nxp.com>
Date: Thu, 21 Mar 2024 15:24:06 +0800
Subject: [PATCH 3/5] Optimize the code of enet_fec driver

enet_fec_dev_init will invoke enet_fec_alloc_buffer directly

Signed-off-by: Hongbo Wang <hongbo.wang@nxp.com>
---
 devices/enet_fec/enet_fec.c  | 6 +++++-
 devices/enet_fec/enet_fec.h  | 1 -
 devices/enet_fec/enet_main.c | 4 ----
 3 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/devices/enet_fec/enet_fec.c b/devices/enet_fec/enet_fec.c
index 73595e68..5bafec4e 100644
--- a/devices/enet_fec/enet_fec.c
+++ b/devices/enet_fec/enet_fec.c
@@ -270,7 +270,7 @@ struct enet_fec_bd_buf_t *enet_fec_get_bd_buf(void)
 	return &senet_fec_bd_buf;
 }
 
-int enet_fec_alloc_buffer(void)
+static int enet_fec_alloc_buffer(void)
 {
 	int ret = 0;
 
@@ -313,6 +313,10 @@ int enet_fec_dev_init(void)
 	pmd_enetfec_probe(&sfec_vdev);
 	dev = sfec_vdev.dev;
 
+	ret = enet_fec_alloc_buffer();
+	if (ret)
+	      goto err_tag;
+
 	enetfec_eth_configure(dev);
 
 	/* init one RX queue */
diff --git a/devices/enet_fec/enet_fec.h b/devices/enet_fec/enet_fec.h
index 580ca291..440f401b 100644
--- a/devices/enet_fec/enet_fec.h
+++ b/devices/enet_fec/enet_fec.h
@@ -35,7 +35,6 @@ typedef uint64_t phys_addr_t;
 
 phys_addr_t rte_mem_virt2phy(const void *virtaddr);
 
-int enet_fec_alloc_buffer(void);
 uint8_t *enet_fec_get_tx_buf(int idx);
 struct enet_fec_bd_buf_t *enet_fec_get_bd_buf(void);
 
diff --git a/devices/enet_fec/enet_main.c b/devices/enet_fec/enet_main.c
index c0e4121e..25e4efbc 100644
--- a/devices/enet_fec/enet_main.c
+++ b/devices/enet_fec/enet_main.c
@@ -238,10 +238,6 @@ int ec_fec_net_init(uint32_t cycle_ns, uint32_t core_mask)
 
     ecus_set_core_mask(core_mask);
 
-    ret = enet_fec_alloc_buffer();  /* should be called before enet_fec_dev_init */
-    if (ret)
-        goto err_tag;
-
     ret = enet_fec_dev_init();
     if (ret)
         goto err_tag;
-- 
2.34.1

